### PR TITLE
RNMT-3824 Errors in the first reads/writes of a Key Store when multiple initializations run simultaneously

### PR DESCRIPTION
**Pull request by usernuno**
## Description
Removes the previous fix.
Improves mutual exclusion in concurrent init calls.

## Context

Fixes https://outsystemsrd.atlassian.net/browse/RNMT-3824

Simultaneous init calls were not being handled correctly. We added proper mutual exclusion and a queue that guarantees only one intent is handled at a time to prevent intent cancellation.

Please review the 2 commits separately for maximum comfort.

## Type of changes

- [ ] Feature (change which adds functionality)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to new feature)
- [x] Fix (change which fixes an issue)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to fix)
- [ ] Performance (change which improves performance)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to performance improvement)
- [ ] Refactor (non-breaking change that is neither feature, fix nor performance)
- [ ] Style (non-breaking change that only affects formatting and/or white-space)

## Components affected
- [x] Android platform
- [ ] iOS platform
- [ ] JavaScript
- [ ] OutSystems

## Tests

Manual tests were performed using Android 9 (uses old intent) and 10 (uses new intent).

## Checklist

- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [ ] Tests have been created
- [x] Code follows code style of this project
- [ ] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

### CODE REVIEW

**Review by luissilvaos — APPROVED** [review]

File: src/android/IntentRequest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package com.crypho.plugins;
+
+import android.content.Intent;
+
+import org.apache.cordova.CallbackContext;
+
+class IntentRequest {
+
+    private IntentRequestType type;
```
Comment (EiyuuZack) [inline src/android/IntentRequest.java:9]: Keeping consistency with other classes and the fact there are only getters present, it appears to me these variables could be declared `final`, but I nitpick.

File: src/android/SecureStorage.java [hunk 2]
```diff
@@ -247,69 +204,106 @@ public void run() {
...
+        }
     }
 
-    private Context getContext() {
-        return cordova.getActivity().getApplicationContext();
+    private void completeInit(final String service, final CallbackContext callbackContext) {
+        cordova.getThreadPool().execute(new Runnable() {
+            public void run() {
+                try {
+                    // RSA already has mutual exclusion in all its public methods individually
```
Comment (EiyuuZack) [inline src/android/SecureStorage.java:279]: 👍

**Review by EiyuuZack — APPROVED** [review]

**Review by sandrarodrigues76 — CHANGES_REQUESTED** [review]
not a single automatic test?

**Review by sandrarodrigues76 — APPROVED** [review]
The automatic test will be handle soon.

**Pull request merged**